doc/Language/classtut.pod6

```diff
@@ -596,57 +501,81 @@ method new(&callback, *@dependencies) {
-do this, we override the C<BUILD> submethod, which is called on the brand
-new object by C<bless>:
+=head2 C<BUILD>
 
-=begin code :preamble<has &.callback; has @.dependencies;>
-submethod BUILD(:&!callback, :@!dependencies) { }
-=end code
+It is possible to disable the automatic attribute initialization and perform the initialization of
+attributes oneself. To do so one needs to write a custom C<BUILD> submethod. There are several edge cases one
+needs to be aware of and take into account though. This is detailed in the
+L<Object Construction Reference|/language/objects#Object_construction>. Because of the difficulty of using
+C<BUILD>, it is recommended to only make use of it when none of the other approaches described above suffices.
 
-Since C<BUILD> runs in the context of the newly created C<Task> object, it
-is allowed to manipulate those private attributes. The trick here is that
-the private attributes (C<&!callback> and C<@!dependencies>) are being used
-as the bind targets for C<BUILD>'s parameters. Zero-boilerplate
-initialization! See L<objects|/language/objects#Object_construction> for
-more information.
+X<|DESTROY>
+X<|submethod DESTROY>
+=head1 Destruction
+
+Raku is a garbage collecting language. This means that one usually doesn't need to care about cleaning up objects,
+because Raku does so automatically. Raku does not give any guarentees as to when it will
+clean up a given object though. It usually does a cleanup run only if the runtime needs the memory, so we
+can't rely on when it's going to happen.
 
-The C<BUILD> method is responsible for initializing all attributes and must
-also handle default values:
+To run custom code when an object is cleaned up one can use the C<DESTROY> submethod. It can for example be used to close handles or supplies or delete temporary files that are no longer g
+oing to be used. As garbage collection can happen at arbitrary points during the runtime of our program, even in the middle of some totally unrelated piece of code in a different thread, we
+ must make sure to not assume any context in our C<DESTROY> submethod.
 
 =begin code
-has &!callback;
-has @!dependencies;
-has Bool ($.done, $.ready);
-submethod BUILD(
-        :&!callback,
-        :@!dependencies,
-        :$!done = False,
-        :$!ready = not @!dependencies,
-    ) { }
+my $in_destructor = 0;
+
+class Foo {
+    submethod DESTROY { $in_destructor++ }
+}
+
+my $foo;
+for 1 .. 6000 {
+    $foo = Foo.new();
+}
+
+say "DESTROY called $in_destructor times";
 =end code
 
-See L<Object Construction|/language/objects#Object_construction> for more
-options to influence object construction and attribute initialization.
+This might print something like C<DESTROY called 5701 times> and possibly only kicks
+in after we have stomped over former instances of C<Foo> a few thousand times. We also can't
+rely, on the order of destruction.
+
+Same as C<TWEAK>: Make sure to always declare C<DESTROY> as a C<submethod>.
```

Comment:
The `DESTROY` section was moved here from above the `Starting with class` section in https://github.com/Raku/doc/pull/4047/commits/d9c784dc392b8e98a337e0eebfca017ed14624d4.